intermediate_source/quadric_layers_tutorial.py

```diff
@@ -0,0 +1,254 @@
+                      Quadric(16, 10),
+                      nn.LogSoftmax(dim=1))
+
+# total number of model parameters
+sum(p.numel() for p in model.parameters())
+
+######################################################################
+# 4. Train the model
+# ------------------
+
+loss_fn = nn.NLLLoss()
+optimizer = optim.SGD(model.parameters(), lr=0.003, momentum=0.9)
+epochs = 20
+for e in range(epochs):
+    train_loss = 0
+    for images, labels in train_loader:
+        images = images.view(images.shape[0], -1)
+        optimizer.zero_grad()
+        output = model(images)
+        loss = loss_fn(output, labels)
+        loss.backward()
+        optimizer.step()
+        train_loss += loss.item()
+    else:
+        print(f"Epoch {e} - Training loss: {train_loss/len(train_loader)}")
+
+######################################################################
+# 5. Evaluate the model
+# ---------------------
+
+# training data
+corrects, all = 0, 0
+for images,labels in train_loader:
+  for i in range(len(labels)):
+    img = images[i].view(1, 784)
+    with torch.inference_mode():
+        logps = model(img)
+    ps = torch.exp(logps)
+    prob = list(ps.numpy()[0])
+    inf_label = prob.index(max(prob))
+    true_label = labels.numpy()[i]
+    if(true_label == inf_label):
+      corrects += 1
+    all += 1
+
+print(f"Number of training images = {all}")
+print(f"\nModel training accuracy = {corrects / all}")
+
+# test data
+corrects, all = 0, 0
+for images,labels in test_loader:
+  for i in range(len(labels)):
+    img = images[i].view(1, 784)
+    with torch.inference_mode():
+        logps = model(img)
+    ps = torch.exp(logps)
+    prob = list(ps.numpy()[0])
+    inf_label = prob.index(max(prob))
+    true_label = labels.numpy()[i]
+    if(true_label == inf_label):
+      corrects += 1
+    all += 1
+
+print(f"Number of test images = {all}")
+print(f"\nModel test accuracy = {corrects / all}")
+
+######################################################################
+# 6. Conclusion
+# -------------
+#
+# Quadric layers can easily be used to reduce model size in many applications just by replacing linear layers.
+#
+# It is hard to quantify model size reduction because of several factors including number of epochs, training and test accuracy etc.
+#
+# In this example, with the same number of epochs and identical training and test data and virtually the same accuracy 
+# a model reduction size of ~75% could be achieved
+#
+# quadric model:
+# --------------
+# model = nn.Sequential(Quadric(784, 16),
+#                      nn.ReLU(),
+#                      Quadric(16, 10),
+#                      nn.LogSoftmax(dim=1)
+# model size: 25434 parameters
+#
+# comparable linear  model:
+# -------------------------
+# model = nn.Sequential(nn.Linear(784, 128),
+#                      nn.ReLU(),
+#                      nn.Linear(128, 10),
+#                      nn.LogSoftmax(dim=1))
+# model size: 101770 parameters
+#
```

Comment:
Please add a conclusion section. For example:

Conclusion
-------------

In this tutorial, you have learned .....<add more text here>
For more information, see: 

* link1
* link2
* link3